src/openrct2/ride/VehicleData.h

```diff
@@ -11,9 +11,23 @@
 
 #include "../common.h"
 
-extern const uint8_t* Rotation1TimeToSpriteMaps[];
-extern const uint8_t* Rotation2TimeToSpriteMaps[];
-extern const uint8_t* Rotation3TimeToSpriteMaps[];
+struct RotationSubtypeSpriteMap
+{
+    const uint8_t* Animations[3] = { 0, 0, 0 };
+    constexpr RotationSubtypeSpriteMap(
+        const uint8_t* accelerationAnimation, const uint8_t* continuousAnimation, const uint8_t* decelerationAnimation)
+    {
+        Animations[0] = accelerationAnimation;
+        Animations[1] = continuousAnimation;
+        Animations[2] = decelerationAnimation;
+    }
+    const uint8_t* GetAnimation(uint8_t sub_state) const
+    {
+        return Animations[sub_state];
+    }
```

Comment:
I think this can be made more C++zy by using `std::array` and constructor initializer lists, maybe even adding an `assert` to make sure we're accessing a valid index.

```suggestion
struct RotationSubtypeSpriteMap
{
    const std::array<const uint8_t*, 3> Animations;
    constexpr RotationSubtypeSpriteMap(
        const uint8_t* accelerationAnimation, const uint8_t* continuousAnimation, const uint8_t* decelerationAnimation)
        : Animations({ accelerationAnimation, continuousAnimation, decelerationAnimation })
    {
    }
    const uint8_t* GetAnimation(uint8_t sub_state) const
    {
        assert(sub_state < Animations.size());
        return Animations.at(sub_state);
    }
};
```

Another idea would be to defined a `constexpr` of the number of valid sub_states. Use that to choose the size of the `std::array` and to write `static_asserts` that the `enum`s controlling the substates have the same number of elements as this `constexpr` is, so compilation would fail if that constant ever changed without changing the enums and vice versa.